litestar/middleware/_utils.py

```diff
@@ -32,10 +32,11 @@ def build_exclude_path_pattern(
         return None
 
     try:
-        pattern = re.compile("|".join(exclude)) if isinstance(exclude, list) else re.compile(exclude)
-        if pattern.match("/"):
-            warn_middleware_excluded_on_all_routes(pattern, middleware_cls=middleware_cls)
-        return pattern
+        exclude_list = [exclude] if isinstance(exclude, str) else exclude
+        for el in exclude_list:
+            if el == "/":
```

Comment:
Hm. I think this has the opposite problem of being too lax, as it would e.g. not warn for `/*`. Maybe we extend the original pattern test with another check to verify it's actually greedy, i.e. matches `/` *and* other paths with a leading `/`?

Reply:
yes you're right
I reverted and kept the tests, but cant find now a good way to match / and other paths.
I just addded a weird test on the pattern being not ^/$ but finding greedy patterns is context dependant so I'm not sure now how to handle this properly of if it even possible.

Reply:
Yeah, I don't think it's possible to do this 100% reliably. But it is just a warning after all, so maybe that's okay. How about testing if the pattern matches `/` *and* `/` followed by something randomly unique? That way we could be somewhat certain that it would greedily match?